pgx-macros/src/lib.rs

```diff
@@ -890,6 +900,7 @@ fn impl_guc_enum(ast: DeriveInput) -> proc_macro2::TokenStream {
 enum PostgresTypeAttribute {
     InOutFuncs,
     PgVarlenaInOutFuncs,
+    FilterDatum,
```

Comment:
Is there a reason this would only apply as a modified to the Default behavior and not possibly a modifier on all InOutFuncs?

Reply:
My thinking was that, if you were implementing your own InOutFuncs, you'd be able to do the filtering in that codeyourself.  However, I have very limited experience with using InOutFuncs, so if that's not the case, I can see about adding filtering in all cases.